.github/workflows/remove-wheels.yml

```diff
@@ -0,0 +1,78 @@
+name: Remove old wheels
+
+on:
+  # Run daily at 1:23 UTC
+  schedule:
+  - cron:  '23 1 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Needed for micromamba pickup
+defaults:
+  run:
+    shell: bash -l {0}
+
+env:
+  N_LATEST_UPLOADS: 5
+  ANACONDA_USER: "scientific-python-nightly-wheels"
+
+jobs:
+  remove:
+
+    runs-on: ubuntu-latest
+    environment:
+      name: remove-old-wheels
+
+    steps:
+      - name: Install micromamba and anaconda-client
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: remove-wheels
+          create-args: >-
+            python=3.11
+            anaconda-client=1.11.2
+
+      - name: Show environment
+        run: env
+
+      - name: Show CLI API info
+        run: |
+          anaconda show --help
+          echo ""
+          anaconda remove --help
+
+      - name: Query package index for packages
+        run: |
+          anaconda show "${ANACONDA_USER}" &> >(grep "${ANACONDA_USER}/") | \
+              awk '{print $1}' | \
+              sed 's|.*/||g' > package-names.txt
+
+      - name: Remove old uploads to save space
+        run: |
+          # Remove all _but_ the last ${N_LATEST_UPLOADS} package versions
+          # N.B.: `anaconda show` places the newest packages at the bottom of the output
+          # of the 'Versions' section and package versions are preceded with a '   + '.
+
+          if [ -s package-names.txt ]; then
+              while LANG=C IFS= read -r package_name ; do
+
+                  printf "\n# package: %s\n" "${package_name}"
```

Comment:
The use of `printf` is my go-to for a 1-liner that allows for visual separation of print information to avoid things getting smushed together in the logs. E.g. the `\n` helps quite a bit IMO with output like

```
# package: ipython

# package: matplotlib
# Removing scientific-python-nightly-wheels/matplotlib/3.6.0.dev3181+g9f17f3f851
# Removing scientific-python-nightly-wheels/matplotlib/3.6.0.dev3182+gd6bad5f684

# package: networkx

# package: numpy
# Removing scientific-python-nightly-wheels/numpy/1.25.0.dev0+1532.g5d6c744db

# package: scikit-image

# package: scikit-learn

# package: scipy

# package: xarray
# Removing scientific-python-nightly-wheels/xarray/2023.5.1.dev9+g609a9016
# Removing scientific-python-nightly-wheels/xarray/2023.5.1.dev10+gf45eb733
# Removing scientific-python-nightly-wheels/xarray/2023.5.1.dev11+gfcb81756
```

Though to make things less visually strange with switching back and forth between `echo` and `printf` I can unify things by just using `echo -e`. :+1: